client/me/memberships/subscription.jsx

```diff
@@ -11,114 +12,124 @@ import HeaderCake from 'calypso/components/header-cake';
 import { withLocalizedMoment } from 'calypso/components/localized-moment';
 import Main from 'calypso/components/main';
 import Notice from 'calypso/components/notice';
+import { CALYPSO_CONTACT } from 'calypso/lib/url/support';
 import titles from 'calypso/me/purchases/titles';
 import MeSidebarNavigation from 'calypso/me/sidebar-navigation';
 import { requestSubscriptionStop } from 'calypso/state/memberships/subscriptions/actions';
 import {
 	getSubscription,
 	getStoppingStatus,
 } from 'calypso/state/memberships/subscriptions/selectors';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
 import { purchasesRoot } from '../purchases/paths';
 import MembershipSiteHeader from './header';
 
 import './subscription.scss';
 
-class Subscription extends React.Component {
-	stopSubscription = () => this.props.requestSubscriptionStop( this.props.subscription.ID );
+function Subscription( { translate, subscription, moment, stoppingStatus } ) {
+	const dispatch = useDispatch();
 
-	render() {
-		const { translate, subscription, moment, stoppingStatus } = this.props;
+	const stopSubscription = () => dispatch( requestSubscriptionStop( subscription.ID ) );
 
-		return (
-			<Main wideLayout className="memberships__subscription">
-				<DocumentHead title={ translate( 'Subscription Details' ) } />
-				<MeSidebarNavigation />
-				<QueryMembershipsSubscriptions />
-				<FormattedHeader brandFont headerText={ titles.sectionTitle } align="left" />
-				<HeaderCake backHref={ purchasesRoot }>{ translate( 'Subscription Details' ) }</HeaderCake>
-				{ stoppingStatus === 'start' && (
-					<Notice
-						status="is-info"
-						isLoading={ true }
-						text={ translate( 'Stopping this subscription' ) }
-					/>
-				) }
-				{ stoppingStatus === 'fail' && (
-					<Notice
-						status="is-error"
-						text={ translate( 'There was a problem while stopping your subscription' ) }
-					/>
-				) }
-				{ stoppingStatus === 'success' && (
-					<Notice
-						status="is-success"
-						text={ translate( 'This subscription has been stopped. You will not be charged.' ) }
-					/>
-				) }
+	useEffect( () => {
+		if ( stoppingStatus === 'fail' ) {
+			// run is-error notice to contact support
+			dispatch(
+				errorNotice(
+					translate(
+						'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
+						{
+							components: {
+								a: <a href={ CALYPSO_CONTACT } />,
+								strong: <strong />,
+							},
+						}
+					)
+				)
+			);
+		} else if ( stoppingStatus === 'success' ) {
+			// redirect back to Purchases list
+			dispatch(
+				successNotice(
+					translate( 'This subscription has been stopped. You will not be charged.' ),
```

Comment:
I find the use of "stopped" versus "cancelled" a bit odd. Since we use "cancel" everywhere else, should we make these unified with that existing standard? Also, I would consider clarifying that you would not be charged in the future; "You will not be charged" could be misconstrued as their *current* charge would be wiped out (which does not appear to be the case).

Reply:
Good points, I can add this in as it does make it more clear. Since it is only used in this error it shouldn't be too big of an issue.

Reply:
Worth noting that this string should be much more common than the other one, so now that this one is changed we should definitely wait for the translations to complete before merging; or alternatively use `hasTranslation` to use the old string as a fallback.

Reply:
@sirbrillig Can you elaborate on what you mean by 'this string _should be_ much more common than the other one?' 

The revised string would be 'This subscription has been canceled. You will no longer be charged.' once finalized.

Reply:
I just mean that this message is displayed when the cancelation is successful, which is the happy path that comes from using this page, so it's likely to be displayed more often than the failure message above. I commented above that we might not need to wait for the full translation of the failure message since it's not likely to be displayed very often, but I don't think that reasoning holds true for the translation of the success message.